### PR TITLE
Synchronization flow

### PR DESCRIPTION
**Pull request by Berardpi**
Rework architecure with one Entity master class and light subclass for each entity.

### CODE REVIEW

File: app/models/entity.rb [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    # if last_synchronization
+      # Cannot get the get_updated query to work
+      # client.get_updated('Account', last_synchronization.updated_at, Time.now)
+      # client.query('select Id, Name from Account ORDER BY Name')
+    # else
+      fields = self.mapping.values.join(', ')
+      entities = client.query("select Id, LastModifiedDate, #{fields} from #{self.external_entity_name} ORDER BY LastModifiedDate DESC")
+      entities = entities.to_a
+
+      index = 0
```
Comment (BrunoChauvet) [inline app/models/entity.rb:49]:
``` ruby
entities.select{|e| e.LastModifiedDate >= last_synchronization.updated_at}
```

File: app/models/entity.rb [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+  def create_entity_to_connec(client, external_entity)
+    response = client.post("/#{self.connec_entity_name.downcase.pluralize}", { "#{self.connec_entity_name.downcase.pluralize}": data_to_connec(external_entity) })
+    JSON.parse(response.body)["#{self.connec_entity_name.downcase.pluralize}"]
+  end
+
+  def update_entity_to_connec(client, external_entity, connec_id)
+    data = data_to_connec(external_entity)
+    data['id'] = connec_id
+    response = client.post("/#{self.connec_entity_name.downcase.pluralize}", { "#{self.connec_entity_name.downcase.pluralize}": data })
```
Comment (BrunoChauvet) [inline app/models/entity.rb]: To update a resource on Connec! API, you need to send a PUT request against the endpoint `resources/id`
For instance:
`PUT /organizations/connec_id`

File: app/models/entity.rb (hunk 1 above)
Comment (Berardpi) [inline app/models/entity.rb:49]: The idea is that the array is sorted so we just have to find the first entity with a LastModifiedDate >= last_synchronization.updated_at.
There's probably a prettier way to do it, but i'm affraid select will uselessly iterate over the whole array (which can be quite long)

File: app/models/entity.rb (hunk 2 above)
Comment (Berardpi) [inline app/models/entity.rb]: Ahhh. That's why it doesn't work.. thanks!

**Pull request merged**